scripts/linters/ts_ignore_exceptions.json

```diff
@@ -0,0 +1,102 @@
+    "spyOn($, 'ajax').and.returnValue(Promise.resolve("
+  ],
+  "core/templates/services/date-time-format.service.spec.ts": [
+    "spyOn(window, 'Date').and.callFake(MockDateContructor);"
+  ],
+  "core/templates/domain/statistics/PlaythroughObjectFactorySpec.ts": [
+    "expect(() => pof.createFromBackendDict(playthroughDict)).toThrowError("
+  ],
+  "extensions/interactions/EndExploration/directives/end-exploration-validation.service.spec.ts": [
+    "customizationArguments.recommendedExplorationIds.value = [1];",
+    "customizationArguments.recommendedExplorationIds.value = 'ExpID0';"
+  ],
+  "extensions/interactions/GraphInput/directives/graph-input-validation.service.spec.ts": [
+    "currentState, {}, answerGroups, goodDefaultOutcome);"
+  ],
+  "core/templates/domain/statistics/LearnerActionObjectFactorySpec.ts": [
+    "learnerActionObjectFactory.createFromBackendDict(playthroughDict);"
+  ],
+  "core/templates/pages/exploration-editor-page/statistics-tab/charts/pie-chart.component.spec.ts": [
+    "window.google = {};"
+  ],
+  "core/templates/services/speech-synthesis-chunker.service.spec.ts": [
+    "mockSpeechSynthesisUtteran);"
+  ],
+  "core/templates/pages/creator-dashboard-page/modal-templates/upload-activity-modal.controller.spec.ts": [
+    "spyOn(document, 'getElementById').and.callFake(function() {",
+    "spyOn(document, 'getElementById').and.callFake(function() {"
+  ],
+  "core/templates/domain/improvements/SuccessiveIncorrectAnswersTaskObjectFactorySpec.ts": [
+    "task_type: '???',"
+  ],
+  "core/templates/pages/exploration-editor-page/statistics-tab/charts/bar-chart.component.spec.ts": [
+    "window.google = {};"
+  ],
+  "core/templates/pages/exploration-editor-page/modal-templates/post-publish-modal.controller.spec.ts": [
+    "spyOn(window, 'getSelection').and.returnValue({"
+  ],
+  "core/templates/domain/improvements/IneffectiveFeedbackLoopTaskObjectFactorySpec.ts": [
+    "task_type: '???',"
+  ],
+  "extensions/interactions/CodeRepl/directives/code-repl-validation.service.spec.ts": [
+    "customizationArguments.language.value = 1;",
+    "customizationArguments.placeholder.value = 1;",
+    "customizationArguments.preCode.value = 1;",
+    "customizationArguments.postCode.value = 1;"
+  ],
+  "core/templates/services/schema-default-value.service.ts": [
+    "this.logger.error('Invalid schema type: ' + schema.type);"
+  ],
+  "core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.service.spec.ts": [
+    "filterOptions.sort = 'Invalid sort value';"
+  ],
+  "core/templates/domain/improvements/NeedsGuidingResponsesTaskObjectFactorySpec.ts": [
+    "task_type: '???',"
+  ],
+  "core/templates/domain/improvements/HighBounceRateTaskObjectFactorySpec.ts": [
+    "task_type: '???',"
+  ],
+  "core/templates/domain/story/StoryObjectFactorySpec.ts": [
+    "sampleStoryBackendDict);"
+  ],
+  "core/templates/domain/statistics/PlaythroughIssueObjectFactorySpec.ts": [
```

Comment:
I think you can drop spec files from this list if we use ts-expect-error for them. I think that's going to be a common case and it's fine to have that be a pattern. We should still prevent abuse, so perhaps the best approach is to add a separate list that whitelists the specific objects for which we allow ts-expect-error.

In general, these exception lists should be small and comprehensive.

Reply:
Actually, scratch the exception list for ts-expect-error. We use that also for testing validations and I don't think it makes sense to have an exception list for those. I still think that having a check that ts-expect-error is only used in spec files, and dropping spec files from this list if we use ts-expect-error for them, is the way to go.

Reply:
> I still think that having a check that ts-expect-error is only used in spec files, and dropping spec files from this list if we use ts-expect-error for them, is the way to go.

done